clowder-framework/clowder
# File "CREATED" status should prevent user download / notify user why they can't download

## Description
### Problem
Download button is already active for files that are CREATED, but not PROCESSED.

### Approach
* When file status is `CREATED`, disable the Download button and displays a tooltip saying we need to wait
* When file status is `CREATED`, show a banner indicating that file bytes aren't available but metadata and comments are

### How to Test
1. Checkout and run this branch locally
2. Login to Clowder
3. Create a dataset, if you haven't already
4. Upload a file, if you haven't already
5. Exec into `mongo` container: `docker exec -it mongo bash`
6. Within the mongo container, launch the mongo shell: `mongo`
7. Manually set the file's status to `CREATED` - NOTE: this effectively simulates an in-transit file
```python
MongoDB shell version v3.4.17
connecting to: mongodb://127.0.0.1:27017
MongoDB server version: 3.4.17
Server has startup warnings: 
2021-02-17T20:48:34.934+0000 I CONTROL  [initandlisten] 
2021-02-17T20:48:34.934+0000 I CONTROL  [initandlisten] ** WARNING: Access control is not enabled for the database.
2021-02-17T20:48:34.934+0000 I CONTROL  [initandlisten] **          Read and write access to data and configuration is unrestricted.
2021-02-17T20:48:34.934+0000 I CONTROL  [initandlisten] 
> use clowder
switched to db clowder
> db.uploads.findOne({ '_id': ObjectId("602d8d63d72896ee1491a1ff") })
{
	"_id" : ObjectId("602d8d63d72896ee1491a1ff"),
	"_typeHint" : "models.File",
	...
	"status" : "PROCESSED"
}
> db.uploads.updateOne({ '_id': ObjectId("602d8d63d72896ee1491a1ff") }, { '$set': { "status": "PROCESSED" }})
{ "acknowledged" : true, "matchedCount" : 1, "modifiedCount" : 1 }
> db.uploads.findOne({ '_id': ObjectId("602d8d63d72896ee1491a1ff") })
{
	"_id" : ObjectId("602d8d63d72896ee1491a1ff"),
	"_typeHint" : "models.File",
	...
	"status" : "CREATED"
}
```
6. Navigate to the file in Clowder: http://localhost:9000/files/602d8d63d72896ee1491a1ff
    * You should see the Download button is disabled
    * You should see a tooltip when hovering to mouse cursor over the Download button
    * You should see a warning banner notifying the user that the bytes for this file are not yet available

## Review Time Estimate


## Types of changes


## Checklist:
